tests/lib/rust/time/src/times.c

```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) 2024 Linaro LTD */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include <zephyr/kernel.h>
+
+/* Rather than trying to get C enums to match in size with Rust ones, just use a known integet type.
+ */
+enum units {
+	UNIT_FOREVER,
+	UNIT_NO_WAIT,
+	UNIT_DUR_MSEC,
+	UNIT_INST_MSEC,
+};
+
+/* Data handed back from C containing processed time constant values.
+ */
+struct time_entry {
+	const char *name;
+
+	uint32_t units;
+
+	/* Value in the given units. */
+	int64_t uvalue;
+
+	/* Value in ticks. */
+	k_timeout_t value;
+};
+
+const struct time_entry time_entries[] = {
+	{
+		.name = "K_FOREVER",
+		.units = UNIT_FOREVER,
+		.uvalue = 0,
```

Comment:
Should this equal -1?

Reply:
Could probably use a better comment. I should actually just leave the `.uvalue` out, since it won't be used. These are hard coded. It is similar to how the `.value` isn't used with the Duration because it is calculated dynamically.

With these, units kind of indicate a special case, and there is no uvalue, just a value.